tests/test_nhg.py

```diff
@@ -251,11 +253,13 @@ def asic_route_nhg_fvs(k):
             ps.set(route_ipprefix, fvs)
             route_count += 1
 
-        asic_db = dvs.get_asic_db()
-
         # Wait and check ASIC DB the count of nexthop groups used
         asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP", MAX_ECMP_COUNT)
-        asic_routes_count = len(asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"))
+
+        # Wait and check ASIC DB the count of routes
+        # Expect asic_routes_count + MAX_ECMP_COUNT - 1 since 2.2.2.0/24 is already set in the previous test case and counted in asic_routes_count
```

Comment:
>since 2.2.2.0/24 is already set in the previous test case and counted in asic_routes_count [](start = 56, length = 90)

We should prevent inter-testcase dependency. With gtest, it is easy to filter specific testcase to run, so never assume the selection or sequence on testcases.

Reply:
Thanks for the suggestion! Updated the previous test case to remove the inter-testcase dependency. Also refactored `test_route_nhg` test case to use dvslib.